### ISSUE TITLE
The process didn't terminate after called jda.shutdown()

### ISSUE THREAD

**Issue by AlexCai2019** [DV8FromTheWorld/JDA#2455]
### General Troubleshooting

- [X] I have checked for similar issues on the Issue-tracker.
- [X] I have updated to the [latest JDA version](https://ci.dv8tion.net/job/JDA/lastSuccessfulBuild/)
- [X] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

### Expected Behaviour

- Java Version: 17.0.2
- Dependencies: [JDA](https://github.com/DV8FromTheWorld/JDA), [slf4j-nop](https://mvnrepository.com/artifact/org.slf4j/slf4j-nop), [json](https://github.com/stleary/JSON-java)

I called `net.dv8tion.jda.api.JDA.shutdown()` to shutdown the bot. The status of the bot in Discord turned gray which should be offline. But the process didn't stop and the `onShutdown` method didn't execute.
I tried using `shutdownNow()`, but that means I couldn't send an offline message to my bot channel when the `/shutdown` command was executed.
I tried the `awaitShutdown()` example in the [documentation](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/JDA.html#awaitShutdown()) and did some changes:
```java
jda.shutdown();
if (!jda.awaitShutdown(10L, TimeUnit.SECONDS))
{
	jda.shutdownNow();
	jda.awaitShutdown();
}
```
At first it might seem like it went offline completly, but the `onShutdown` method still didn't execute. As soon as I active the bot again, any event will trigger the `onShutdown` event and the bot will go offline again.
This is [my repository](https://github.com/AlexCai2019/Cartoland/tree/8dc04a4dbb462cb167d672556e5a40012e13d37c).

### Code Example for Reproduction Steps

```java
class Event extends ListenerAdapter
{
	@Override
	public void onSlashCommandInteraction(SlashCommandInteractionEvent event)
	{
		if (event.getName().equals("shutdown"))
		{
			event.reply("shutdown").queue();
			Main.jda.shutdown();
		}
	}

	@Override
	public void onShutdown(ShutdownEvent event)
	{
		System.out.println("onShutdown");
	}
}
```

### Code for JDABuilder or DefaultShardManagerBuilder used

```java
public class Main
{
	static JDA jda;
	public static void main(String[] args) throws InterruptedException
	{
		if (args.length < 1)
			return;

		jda = JDABuilder.createDefault(args[0])
				.addEventListeners(new Event())
				.enableIntents(GatewayIntent.MESSAGE_CONTENT, GatewayIntent.GUILD_MEMBERS)
				.setMemberCachePolicy(MemberCachePolicy.ALL)
				.setActivity(Activity.playing("Do /help for more information"))
				.build();

		jda.updateCommands().addCommands(Commands.slash("shutdown", "Use this to shut down the bot")).queue();

		jda.awaitReady();
	}
}
```

### Exception or Error

```yesyes
N/A
```

**Comment by AlexCai2019** [comment]
These are threads that are not daemon:
![image](https://user-images.githubusercontent.com/89138532/234668328-b1c5f2c5-08a3-44f6-9c5e-8f1a312d2ac4.png)

**Comment by MinnDevelopment** [comment]
Are you sure this is the correct JDABuilder code? By default, JDA only creates 5 rate-limiter threads, but your thread dump says 8.

**Comment by MinnDevelopment** [comment]
Could you test if this issue is fixed with #2458? (See [using new features](https://jda.wiki/using-jda/using-new-features/))

**Comment by AlexCai2019** [comment]
> Are you sure this is the correct JDABuilder code? By default, JDA only creates 5 rate-limiter threads, but your thread dump says 8.

I apology for that I want to start with a minimal reproducible example, but I forgot to change my project back to this example when I took this screenshot. Also, I should use text instead of image.
These are non-daemon threads that has "JDA" in their name using the example that I provided (using `jstack -l <pid>`):
```
"JDA RateLimit-Worker 1" #18 prio=5 os_prio=0 cpu=453.12ms elapsed=93.88s tid=0x000001b7541ef560 nid=0x4ab0 waiting on condition  [0x000000360b3fe000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.2/Native Method)
        - parking to wait for  <0x0000000711200170> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.2/LockSupport.java:252)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@17.0.2/AbstractQueuedSynchronizer.java:1672)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(java.base@17.0.2/ScheduledThreadPoolExecutor.java:1218)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(java.base@17.0.2/ScheduledThreadPoolExecutor.java:899)
        at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@17.0.2/ThreadPoolExecutor.java:1061)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.2/ThreadPoolExecutor.java:1122)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.2/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.2/Thread.java:833)

   Locked ownable synchronizers:
        - None
```
```
"JDA RateLimit-Worker 7" #41 prio=5 os_prio=0 cpu=0.00ms elapsed=3.88s tid=0x000001b754a88150 nid=0x1204 waiting on condition  [0x00000036092fe000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.2/Native Method)
        - parking to wait for  <0x0000000711200170> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.2/LockSupport.java:252)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@17.0.2/AbstractQueuedSynchronizer.java:1672)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(java.base@17.0.2/ScheduledThreadPoolExecutor.java:1218)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.poll(java.base@17.0.2/ScheduledThreadPoolExecutor.java:899)
        at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@17.0.2/ThreadPoolExecutor.java:1061)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.2/ThreadPoolExecutor.java:1122)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.2/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.2/Thread.java:833)

   Locked ownable synchronizers:
        - None
```
